### PR TITLE
fix: Fix detection if a gasless DAO can mint, provide testing for new census3

### PR DESCRIPTION
**Pull request by emmdim**
## Description

Fix detection if a gasless dao can mint and provides testing for the new census3

Task: [APP-0000](https://aragonassociation.atlassian.net/browse/APP-0000)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -19,6 +19,7 @@ VITE_FEATURE_FLAG_GASLESS_PLUGIN=true
 VITE_FEATURE_FLAG_HIDE_GOERLIBASED_NETWORKS=true
 
 VITE_VOCDONI_ENV='stg'
+VITE_VOCDONI_CENSUS3_URL=https://census3-aragon.vocdoni.net/api
```
Comment (cgero-eth) [inline .env.development:22]: Do you plan to have a different URL depending on the environment?

File: package.json [hunk 2]
```diff
@@ -36,9 +36,9 @@
     "@tiptap/pm": "^2.0.3",
     "@tiptap/react": "^2.0.3",
     "@tiptap/starter-kit": "^2.0.3",
-    "@vocdoni/gasless-voting": "0.0.1-rc27",
+    "@vocdoni/gasless-voting": "./vocdoni-gasless-voting-v0.0.1-rc27-7.tgz",
```
Comment (cgero-eth) [inline package.json]: Can we avoid pushing the library build to the repository?

File: src/hooks/useCensus3.tsx [hunk 3]
```diff
@@ -11,7 +11,9 @@ import {useParams} from 'react-router-dom';
 import {useProposal} from '../services/aragon-sdk/queries/use-proposal';
 import {GaslessVotingProposal} from '@vocdoni/gasless-voting';
 
-const CENSUS3_URL = 'https://census3-stg.vocdoni.net/api';
+export const CENSUS3_URL =
+  import.meta.env.VITE_VOCDONI_CENSUS3_URL ??
```
Comment (cgero-eth) [inline src/hooks/useCensus3.tsx:15]: Currently the URL is defined for all the environments, we don't need a fallback

File: src/services/vocdoni-census3/queries/use-census3-members.ts [hunk 4]
```diff
@@ -28,15 +28,21 @@ export const useCensus3Members = (
...
-    });
+    return census3
+      .getStrategyHolders(strategyId!, {
+        pageSize: page || -1,
+      })
+      .then(holders => holders)
+      .catch(error => {
+        console.debug(error);
+        return null;
+      });
```
Comment (cgero-eth) [inline src/services/vocdoni-census3/queries/use-census3-members.ts:39]: The `.then(holders => holders)` is redundant, right? Also you can use the `/services/logger` utility class to log errors, these are sent to Sentry for us to investigate on Application issues

File: src/context/createGaslessProposal.tsx [hunk 5]
```diff
@@ -199,6 +199,8 @@ const useCreateGaslessProposal = (daoTokenAddress: string | undefined) => {
       const censusToken = await getCensus3Token();
 
       // Create the vocdoni census
+      census3.queueWait.retryTime = 5000;
```
Comment (cgero-eth) [inline src/context/createGaslessProposal.tsx:202]: We updated the whole transaction flow for the create-dao and create-proposal processes, could you update your branch to use the latest updates on develop?

File: src/services/aragon-sdk/queries/use-voting-settings.ts [hunk 6]
```diff
@@ -26,9 +37,56 @@ async function fetchVotingSettingsAsync(
     blockNumber
   );
 
+  if (
```
Comment (cgero-eth) [inline src/services/aragon-sdk/queries/use-voting-settings.ts:40]: Can we introduce this data fetching to the `useGaslessGovernanceEnabled` hook? We could implement a new hook like `useHasPermission` to fetch the data and use it on the `useGaslessGovernanceEnabled` hook

**Review by cgero-eth — COMMENTED** [review]
Thank you for the fix @emmdim!

File: package.json (hunk 2 above)
Comment (emmdim) [reply package.json]: @cgero-eth  for sure, itis while you test the funcionality. If everything is ok I will publish the package and update the PR :-)

File: src/hooks/useCensus3.tsx (hunk 3 above)
Comment (emmdim) [reply src/hooks/useCensus3.tsx:15]: The URL below, in staging is the fallback URL. Is that what you mean?

File: .env.development (hunk 1 above)
Comment (emmdim) [reply .env.development:22]: It is possible, we can work in this in the future, but currently we would like to test this new dedicated service :-)

File: src/services/aragon-sdk/queries/use-voting-settings.ts (hunk 6 above)
Comment (emmdim) [reply src/services/aragon-sdk/queries/use-voting-settings.ts:40]: OK. I can do it. But which solution you prefere @cgero-eth ?

File: src/hooks/useCensus3.tsx (hunk 3 above)
Comment (cgero-eth) [reply src/hooks/useCensus3.tsx:15]: Mm.. If you want to use the staging url on the staging environment, you should set this (https://census3-stg.vocdoni.net/api) on the .env.staging file and remove the fallback here to:
```
export const CENSUS3_URL = import.meta.env.VITE_VOCDONI_CENSUS3_URL!
```

**Comment by emmdim** [comment]
Continued here and resolved the issues here:
https://github.com/aragon/app/pull/1334
@cgero-eth

**Pull request closed without merging**